File: BeagleBone/Black/blinkLED.js
#!/usr/bin/env nodejs
////////////////////////////////////////
//	blinkLED.js
//      Blinks the USR LEDs and P9_14.
//	Wiring:	P9_14 connects to the plus lead of an LED.  The negative lead of the
//			LED goes to a 220 Ohm resistor.  The other lead of the resistor goes
//			to ground.
//	Setup:	
//	See:	
////////////////////////////////////////
const fs = require("fs");

// Look up P9.14 using show-pins.  gpio1.18 maps to 50
pin="50";
pin_in = "51";

GPIOPATH="/sys/class/gpio/";
// Make sure pin is exported
if(!fs.existsSync(GPIOPATH+"gpio"+pin)) {
    fs.writeFileSync(GPIOPATH+"export", pin);
}
if(!fs.existsSync(GPIOPATH+"gpio"+pin_in)) {
    fs.writeFileSync(GPIOPATH+"export", pin_in);
}
// Make it an output pin
fs.writeFileSync(GPIOPATH+"gpio"+pin+"/direction", "out");
// Make it an input pin
fs.writeFileSync(GPIOPATH+"gpio"+pin_in+"/direction", "in");

// Blink every 500ms
setInterval(toggle, 500);

state="1";
function toggle() {
    fs.writeFileSync(GPIOPATH+"gpio"+pin+"/value", state);
    in_status = fs.readFileSync(GPIOPATH+"gpio"+pin_in+"/value");
    if(state == "0") {
        state = "1";
    } else {
        state = "0";
    }
    
}

File: BeagleBone/Black/blinkLED.py
#!/usr/bin/python
#//////////////////////////////////////
#	blink.py
#	Blinks one LED wired to P9_14.
#	Wiring:	P9_14 connects to the plus lead of an LED.  The negative lead of the
#			LED goes to a 220 Ohm resistor.  The other lead of the resistor goes
#			to ground.
#	Setup:	
#	See:	
#//////////////////////////////////////
import time
import os

# Look up P9.14 using show-pins.  gpio1.18 maps to 50
pin = "51"
 
GPIOPATH='/sys/class/gpio/'
# Make sure pin is exported
if (not os.path.exists(GPIOPATH+"gpio"+pin)):
    f = open(GPIOPATH+"export", "w")
    f.write(pin)
    f.close()

# Make it an output pin
f = open(GPIOPATH+"gpio"+pin+"/direction", "w")
f.write("out")
f.close()
 
f = open(GPIOPATH+"gpio"+pin+"/value", "w")
# Blink
while True:
    f.seek(0)
    f.write("1")
    time.sleep(0.5)

    f.seek(0)
    f.write("0")
    time.sleep(0.5)
f.close()